### DISCUSSION TITLE
Dynamic imported media in getStaticProps does not generate actual bundle/media file

### DISCUSSION THREAD

**Discussion by skirsten** [vercel/next.js#36621] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

    Operating System:
      Platform: linux
      Arch: x64
      Version: #112-Ubuntu SMP Thu Feb 3 13:50:55 UTC 2022
    Binaries:
      Node: 14.18.1
      npm: 6.14.15
      Yarn: 1.22.17
      pnpm: N/A
    Relevant packages:
      next: 12.1.6-canary.15
      react: 18.1.0
      react-dom: 18.1.0

### Describe the Bug

Dynamic imports in `getStaticProps` e.g.

```ts
const { default: image } = await import(`../img/a.jpg`);
const { default: image } = await import(`../img/${slug}.jpg`);
```

work as expected:
```json 
{
  "src": "/_next/static/media/a.30e618ce.jpg",
  "height": 300,
  "width": 200,
  "blurDataURL": "/_next/image?url=%2F_next%2Fstatic%2Fmedia%2Fa.30e618ce.jpg&w=8&q=70"
}
```

but the actual file this points to does not exist:
```
curl --head localhost:3000/_next/static/media/a.30e618ce.jpg
HTTP/1.1 404 Not Found
```

### Expected Behavior

The bundled media file (`/_next/static/media/a.30e618ce.jpg` in this case) exists and can be served

### To Reproduce

https://codesandbox.io/s/intelligent-nightingale-1994wu?file=/pages/%5Bslug%5D.tsx:453-583

**Comment by balazsorban44** [comment]
Please refer to the docs:

> Dynamic await import() or require() are not supported. The import must be static so it can be analyzed at build time.

https://nextjs.org/docs/basic-features/image-optimization#local-images

Also note:

> Only assets that are in the public directory at [build time](https://nextjs.org/docs/api-reference/cli#build) will be served by Next.js. Files added at runtime won't be available. We recommend using a third party service like [AWS S3](https://aws.amazon.com/s3/) for persistent file storage.

https://nextjs.org/docs/basic-features/static-file-serving
